File: BalmacedaAV/Java3/Leccion1/MundoPc2/src/ar/com/system2023/mundopc/Computadora.java
/*
 * MundoPc
 */
package ar.com.system2023.mundopc;

/*
 * @author Adriana
 */
public class Computadora {
    private final int idComputadora;
    private String nombre;
    private Monitor monitor;
    private Teclado teclado;
    private Raton raton;
    private static int contadorComputadoras;
    
    //Constructor vacio
    private Computadora(){
        this.idComputadora = ++Computadora.contadorComputadoras;
    }
    //Constructor 2
    public Computadora(String nombre){
        
    }
}

File: SiroTorres/Python/Leccion14/mundo_pc/computadora.py
from mundo_pc.monitor import Monitor
from mundo_pc.raton import Raton
from mundo_pc.teclado import Teclado


class Computadora:

    contador_computadoras = 0

    def __init__(self, nombre, monitor, teclado, raton):
        Computadora.contador_computadoras += 1
        self._id_computadoras = Computadora.contador_computadoras
        self._nombre = nombre
        self._monitor = monitor
        self._teclado = teclado
        self._raton = raton

    def __str__(self):
        return f'''
            {self._nombre}: {self._id_computadoras}
            Monitor: {self._monitor}
            Teclado: {self._teclado}
            Raton: {self._raton}
        '''
if __name__ == '__main__':
    teclado1 = Teclado('HP', 'Usb')
    monitor1 = Monitor('Hp', '15 pulgadas')
    raton1 = Raton('HP', 'USB')
    computadora1 = Computadora('HP',monitor1, teclado1, raton1)
    print(computadora1)

    teclado2 = Teclado('Acer', 'Bluetooth')
    monitor2 = Monitor('Acer', '25 pulgadas')
    raton2 = Raton('Acer', 'Bluetooth')
    computadora2 = Computadora('Acer', monitor2, teclado2, raton2)
    print(computadora2)